grpc_android/Dockerfile

```diff
@@ -0,0 +1,73 @@
+# notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#     * Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# Dockerfile for the gRPC Java dev image
+FROM grpc/java_base
+
+# Required by accessing Android's aapt
+RUN apt-get install -y lib32stdc++6 lib32z1
+
+# Install Android SDK 24.2
+RUN cd /usr/local/ && curl -L -O http://dl.google.com/android/android-sdk_r24.2-linux.tgz && tar xf android-sdk_r24.2-linux.tgz
+
+# Environment variables
+ENV ANDROID_HOME /usr/local/android-sdk-linux
+ENV PATH $PATH:$ANDROID_HOME/tools
+ENV PATH $PATH:$ANDROID_HOME/platform-tools
+# Avoid emulator assumes HOME as '/'.
+ENV HOME /root
+
+# Update sdk for android 5.1 (API level 22)
+RUN echo y | android update sdk --all --filter platform-tools,build-tools-22.0.1,sys-img-armeabi-v7a-android-22,android-22,extra-android-m2repository,extra-google-m2repository --no-ui --force
+
+# Create an AVD with API level 22
+RUN echo no | android create avd --force -n avd-api-22 -t android-22
+
+# Pull gRPC Java.
+RUN git clone --depth 1 https://github.com/grpc/grpc-java.git /var/local/git/grpc-java
+
+# Trigger download of needed Maven and Gradle artifacts.
+RUN cd /var/local/git/grpc-java/core && ../gradlew install
+RUN cd /var/local/git/grpc-java/stub && ../gradlew install
+RUN cd /var/local/git/grpc-java/okhttp && ../gradlew install
+RUN cd /var/local/git/grpc-java/protobuf-nano && ../gradlew install
+
+# Pull gRPC Android integration test App
+RUN git clone --depth 1 https://github.com/madongfly/grpc-android-test.git /var/local/git/grpc-android-test
+
+# Config android sdk for gradle
+RUN cd /var/local/git/grpc-android-test && echo "sdk.dir=/usr/local/android-sdk-linux" > local.properties
+
+# Build apks to trigger download of needed Maven and Gradle artifacts.
+RUN cd /var/local/git/grpc-android-test && ./gradlew assemble
```

Comment:
That's a good question, I looked the output of both commands:
```
simonma@simonma-linux:~/github/grpc-android-test$ cat assembleTestOutput | grep "compile"
:app:compileDebugAndroidTestNdk UP-TO-DATE
:app:compileDebugAidl UP-TO-DATE
:app:compileDebugRenderscript UP-TO-DATE
:app:compileDebugJava UP-TO-DATE
:app:compileDebugAndroidTestAidl UP-TO-DATE
:app:compileDebugAndroidTestRenderscript UP-TO-DATE
:app:compileDebugAndroidTestJava UP-TO-DATE

simonma@simonma-linux:~/github/grpc-android-test$ cat assembleOutput | grep "compile"
:app:compileDebugNdk UP-TO-DATE
:app:compileDebugAidl UP-TO-DATE
:app:compileDebugRenderscript UP-TO-DATE
:app:compileDebugJava UP-TO-DATE
:app:compileReleaseAidl UP-TO-DATE
:app:compileReleaseRenderscript UP-TO-DATE
:app:compileReleaseJava UP-TO-DATE
:app:compileReleaseNdk UP-TO-DATE

simonma@simonma-linux:~/github/grpc-android-test$ cat assembleOutput | grep "package"
:app:packageDebug
:app:packageRelease
simonma@simonma-linux:~/github/grpc-android-test$ cat assembleTestOutput | grep "package"
:app:packageDebugAndroidTest UP-TO-DATE
```
It shows the assembleAndroidTest task actually compiled all things needed, it just didn't do a "package" to create the apk.

But we do benefit from this question, that is I found we built the release apk which is not needed, will fix that.